File: src/keep2shareSDK/keep2shareSDK/Authentication.vb
Imports keep2shareSDK.JSON
Imports Newtonsoft.Json

Public Class Authentication


#Region "Get_Token"
    Shared Async Function OneHourToken(Username As String, Password As String, Optional CaptchaWord As String = Nothing, Optional ChallengeID As String = Nothing) As Task(Of String)
        Net.ServicePointManager.Expect100Continue = True : Net.ServicePointManager.SecurityProtocol = Net.SecurityProtocolType.Tls Or Net.SecurityProtocolType.Tls11 Or Net.SecurityProtocolType.Tls12 Or Net.SecurityProtocolType.Ssl3

        Using localHttpClient As New HttpClient(New HCHandler)
            Dim HtpReqMessage As New Net.Http.HttpRequestMessage(Net.Http.HttpMethod.Post, New pUri("login"))
            Dim JSONobj = New With {.username = Username, .password = Password, .captcha_challenge = ChallengeID, .captcha_response = CaptchaWord}
            HtpReqMessage.Content = JSONobj.JsonContent
            Using response As Net.Http.HttpResponseMessage = Await localHttpClient.SendAsync(HtpReqMessage, Net.Http.HttpCompletionOption.ResponseContentRead).ConfigureAwait(False)
                Dim result As String = Await response.Content.ReadAsStringAsync()

                If Linq.JObject.Parse(result).SelectToken("status").ToString = "success" Then
                    Return Linq.JObject.Parse(result).SelectToken("auth_token").ToString
                Else
                    ShowError(result)
                End If
            End Using
        End Using
    End Function
#End Region

End Class

File: keep2shareSDK/keep2shareSDK/Authentication.cs
using Newtonsoft.Json.Linq;
using System.Threading.Tasks;
using static keep2shareSDK.Basic;
using System.Net;
using System.Net.Http;


namespace keep2shareSDK
{
    public class Authentication
    {

        public static async Task<string> OneHourToken(string Username, string Password, string CaptchaWord = null, string ChallengeID = null)
        {
            ServicePointManager.Expect100Continue = true; ServicePointManager.SecurityProtocol = SecurityProtocolType.Tls | SecurityProtocolType.Tls11 | SecurityProtocolType.Tls12 | SecurityProtocolType.Ssl3;

            using (HtpClient localHttpClient = new HtpClient(new HCHandler()))
            {
                HttpRequestMessage HtpReqMessage = new HttpRequestMessage(HttpMethod.Post, new pUri("login"));
                var JSONobj = new { username = Username, password = Password, captcha_challenge = ChallengeID, captcha_response = CaptchaWord };
                HtpReqMessage.Content = JSONobj.JsonContent();
                using (HttpResponseMessage response = await localHttpClient.SendAsync(HtpReqMessage, HttpCompletionOption.ResponseContentRead).ConfigureAwait(false))
                {
                    string result = await response.Content.ReadAsStringAsync();

                    if (JObject.Parse(result).SelectToken("status").ToString() == "success")
                    {
                        return JObject.Parse(result).SelectToken("auth_token").ToString();
                    }
                    else
                    {
                        ShowError(result);
                        return null;
                    }
                }
            }
        }
    }


}
